utils/scarb-build-metadata/build.rs

```diff
@@ -52,7 +52,7 @@ fn cairo_version() {
         .root
         .expect("Expected metadata resolve root to be present.");
     assert!(
-        root.repr.starts_with("scarb "),
+        root.repr.contains("scarb#"),
```

Comment:
The change done this way effectively sets our MRSV to 1.77, so we should set appropriate field in our Cargo.tomls. 

@fracek are you able to use Rust 1.77 in Nix pkgs ?

Reply:
Maybe we could just check for either `starts_with("scarb ")` or `contains("scarb#")`

Reply:
Done.